python/ray/experimental/client/server/core_ray_api.py

```diff
@@ -25,13 +26,10 @@ def wait(self, *args, **kwargs):
     def remote(self, *args, **kwargs):
         return ray.remote(*args, **kwargs)
 
-    def call_remote(self, f, *args, **kwargs):
-        return f.remote(*args, **kwargs)
+    def call_remote(self, f: ClientRemoteFunc, *args, **kwargs):
+        if f._raylet_remote_func is None:
+            f._raylet_remote_func = ray.remote(f._func)
```

Comment:
Do we need to apply the remote decorator args here (e.g., num_cpus=1)?

Reply:
Remote decorator args are probably a followup, but yeah, this is where it gets piped in.

Reply:
Can we add this? Should be simple enough.

Reply:
They require another field in the proto -- they tell us how to execute the function, instead of the parameters to execute that function.

Again, it's a simple addition, but orthogonal to the change.

Reply:
Ok. Can you follow-up?

Reply:
#12282 for tracking